packages/components/ovh-shell/src/plugin/ux/components/component.ts

```diff
@@ -0,0 +1,43 @@
+export interface IUxComponentOptions {
+  visible?: boolean;
+}
+
+export interface IUxComponent {
+  toggleVisibility: CallableFunction;
+  show: CallableFunction;
+  hide: CallableFunction;
+  getVisibility: CallableFunction;
```

Comment:
These function have a specific signature and should be defined rather than using `CallableFunction` to define them. 
We use `CallableFunction` when we don't know the signature of the function, like in the PluginManager.